apiary.apib

```diff
@@ -1682,6 +1682,364 @@ asynchronously once the job is created.
             }
 
 
+# Group Factory API
+
+The Factory API allows you to interact with orders that have been placed and move them through the various
+stages of production. The typical workflow is as follows:
+
+<a href="https://github.com/unmadeworks/api-docs/assets/13105806/c638182b-ab00-4141-96de-e4fe77de2a1f">
+  <img src="https://github.com/unmadeworks/api-docs/assets/13105806/c638182b-ab00-4141-96de-e4fe77de2a1f" alt="Image showing the typical user journey through the e-commerce site" width="100%">
+</a>
+
+
+#### Factory Slugs
+
+All Factory API endpoints begin with `/v2/factories/<factory_slug>` where `factory_slug` is the unique identifier
+for each of your factory locations. These slugs will be provided to you by Unmade. Each API token provided to you
+can have access to **one or more individual factory locations**, depending on your specific requirements.
+
+#### Order States
+
+During production, orders progress through the following states:
+
+ - `new`: The order has not yet begun production
+ - `in_production`: The order has been accepted and is now in production
+ - `ready-to-ship`: The order has been produced and is awaiting shipping
+ - `shipped`: The order has left the factory
+ - `cancelled`: The order has been cancelled
+
+#### Job States
+
+After creation, the line items from a given order are split into one or more production jobs. These jobs represent
+one or more individual garments to be manufactured. Each job has its own production state, which are as follows:
+
+ - `new`: The job has not yet begun production
+ - `in_production`: The job has been accepted and is now in production
+ - `ready-to-ship`: The job has been produced and is awaiting shipping
+ - `shipped`: The job has left the factory
+ - `cancelled`: The job has been cancelled
+
+## List Orders [/v2/factories/{factory_slug}/orders{?state,page}]
+
+Orders created using the Order API (see above) will appear in this list after they have been submitted to Unmade
+Factory OMS. You may choose to check this endpoint daily or hourly and process any new orders as necessary.
+
+### List existing orders [GET]
+
++ Parameters
+    + factory_slug (string) - The unique identifier of required factory location
+    + state (optional, string) - Only show orders with this state (see Order States section above for allowed values)
+    + page (optional, number) - Page number
+
++ Request
+    + Headers
+
+            Authorization: Token ABCDEF
+
++ Response 200 (application/json)
+
+    + Body
+
+            {}
+
+## Retrieve Order [/v2/factories/{factory_slug}/order/{order_id}]
+
+If you have been provided with Order IDs directly from another system, you can use this endpoint to retrieve
+data on the order directly. The response from this endpoint contains the same data as the order representation
+in the list endpoint.
```

Comment:
All order IDs in this API must be the Embed Order IDs. Any payloads coming up from Factory will need to have their Factory Order IDs replaced with the Embed one. This will be a pain but there's no way around it really.

Reply:
Sounds good for now. This made me think of what is the difference conceptually between an embed order and a factory order. Embed as a proxy between the client and the factory is becoming more of a nuisance.

Reply:
I did think about if we should just make these new endpoints Factory OMS public endpoints. But that might be even more confusing in the long run. Or maybe it's better? It would certainly be easier for us to implement.

Reply:
I think long run it's better. From a design point of view, you wanna have one source of truth. But yeah, it's beyond the scope of this work I think

Reply:
Maybe we should just put them on Factory now. It would be significantly easier to build. We wouldn't have to do all the faffing with the Embed permissions and tokens: we'd just be handing out Factory OMS tokens and using the existing roles / organisations system for authorization. No proxying needed.

One downside is the different URLs and different tokens vs. the Embed order create API. But maybe it's clear that you have one token for ecom stuff and one token for production stuff.

Reply:
As in, keep order submission on embed, but order read/update on factory directly?

Reply:
Yeah.

Keep `<customer>.embed.unmade.com/v1/orders` how it is now, but expose `factory.unmade.com/v1/orders` to do all the new stuff